ActiveCampaign documentation update (#443)
rudderlabs/rudderstack-docs: destinations/marketing/activecampaign.mdx

</div>

2. Assign a name to your destination and click on **Next**.

### Connection settings

To successfully configure ActiveCampaign as a destination, you will need to configure the following settings:

<img src="../../assets/event-stream-destinations/activecampaign-connection-settings.png" alt="ActiveCampaign connection settings" />

- **API URL:** Your API URL is the unique URL generated against your account. It can be found in your account on the **Settings** page under the **Developer** tab.
- **API Key:** Your API key can be found in your account on the **Settings** page under the **Developer** tab. Each user in your ActiveCampaign account has their own unique API key.
- **Event Key:** This value is unique to your ActiveCampaign account. To obtain the event key, go to your ActiveCampaign account and navigate to **Settings** > **Tracking** > **Event Tracking**.
- **ActID:** This value is unique to your ActiveCampaign account.  Go to **Settings** > **Tracking** > **Event Tracking API**. You will find the **actid** listed here.

## Identify

> The `identify` call lets you associate a user with their actions and capture all the relevant traits about them. This information includes unique `userid` as well as any optional information such as name, email address, etc.

Reviewer: ```suggestion
The `identify` call lets you associate a user with their actions and captures all the relevant traits about them. This information includes unique `userid` as well as any optional information such as name, email address, etc.
```

A sample `identify` call is shown below:

```javascript
rudderanalytics.identify(
  "userId", {
    email: "john@example.com",
    firstName: "John",
    lastName: "Keener,
    phone: "1234567890",
  })
```

In the above snippet, RudderStack captures relevant information about the user such as the `userId` as well as the associated traits such as `email`, `phone`, and the name (`firstName` and `lastName`) of that user.

<div class="infoBlock">

> The <code class="inline-code">email</code> trait is a mandatory trait for mapping a user to ActiveCampaign. If a user already exists, the new values will be updated for that user.

Reviewer: >If a user already exists, the new values will be updated for that user.
bit confusing, does it mean that if a user already exists, the new 'email' value will be updated for that user?

```javascript
rudderanalytics.identify(
  "userId", {
    email: "john@example.com",
    firstName: "John",
    lastName: "Keener,
    phone: "1234567890",
    tags: ["Returning User", "Coupon Used"],
  })
```

<div class="infoBlock">

> The <code class="inline-code">tags</code> property should contain an array of tags which you want to associate with the user. If any tag is already created in the destination previously, RudderStack will automatically skip the creation of that tag.

Reviewer: ```suggestion
The <code class="inline-code">tags</code> property should contain an array of tags which you want to associate with the user. If any tag is already created in the destination previously, RudderStack will override the tag information with the new one.
```

Does it mean this?

<div class="infoBlock">

> The <code class="inline-code">fieldInfo</code> property contains the value of the field information that you want to store for that contact. For using this feature, you have to create the fields from your ActiveCampaign dashboard \(For example - Interest, Country\) before passing the values for the given user.

Reviewer: ```suggestion
The <code class="inline-code">fieldInfo</code> trait contains the value of the custom field that you want to store for that contact. For using this feature, you have to create the custom fields from your ActiveCampaign dashboard \(For example - Interest, Country\) before passing the values for the given user.
```

We have used `fieldInfo` **trait** above. Can 'trait' and 'property' be used interchangeably?

<div class="infoBlock">

For associating a contact to any field, you need to create the list from ActiveCampaign and use the <code class="inline-code">id</code> parameter for mapping that contact.
</div>

## Page

> The `page` call allows you to record information whenever a user sees a web page, along with the associated optional properties of that page. This method must be called at least once per page load.

Reviewer: ```suggestion
The `page` call lets you record information whenever a user sees a web page, along with the associated optional properties of that page. This method must be called at least once per page load.
```

When you call `page`, RudderStack will send that event to ActiveCampaign as a `site tracking event`. This will add your domain to whitelist for tracking purposes.

A sample `page` call looks like the following:

```javascript
rudderanalytics.page("home", {
  path: "path",
  url: "url",
  title: "title",
  search: "search",
  referrer: "referrer",
})
```

In the above sample, RudderStack captures the information related to the page being viewed, the URL `property` is used to whiltelist the website in the destination.

<div class="infoBlock">

A <code class="inline-code">page</code> call will only work if <strong>Site Tracking</strong> is enabled in ActiveCampaign. You can enable this setting by going to the <strong>Tracking</strong> tab in your ActiveCampaign settings page.
</div>

## Screen

> The `screen` method allows you to record whenever a user sees the mobile screen, along with any associated optional properties. This call is similar to the `page` call, but is exclusive to your mobile device.

Reviewer: ```suggestion
The `screen` method lets you record whenever a user sees the mobile screen, along with any associated optional properties. This call is similar to the `page` call, but is exclusive to your mobile device.
```

```objectivec
[[RSClient sharedInstance] screen:@"Sample Screen Viewed" properties:@{@"prop_key" : @"prop_value"}];
```

In the above snippet, RudderStack captures information related to the screen being viewed, along with any additional info on that screen view event.

<div class="warningBlock">

The <code class="inline-code">screen</code> event name must contain only alphanumeric characters. ActiveCampaign rejects events with any special characters in the name.
</div>

## Track

The `track` call lets you capture any user actions and the associated properties. Each action is considered to be an event.

A sample `track` call looks like the following:

```javascript
rudderanalytics.track("Product Purchased", {
  name: "Rubik's Cube",
})
```

In the above snippet, RudderStack captures the information related to the `Product Purchased` event, along with any additional information about that event - in this case the name of the product.

<div class="warningBlock">

The <code class="inline-code">track</code> event name must contain only alphanumeric characters. ActiveCampaign rejects events with any special characters in the name.
</div>

<div class="infoBlock">

> If you want to set a value for the <code class="inline-code">screen</code> and <code class="inline-code">track</code> type event, you need to pass <code class="inline-code">name</code> in the event properties, as shown in the above example.

Reviewer: We have mentioned 'name' only in track-type event, not in screen event but say 'as shown in the above example'. Can be misleading..

</div>

## FAQ

### Can I use special characters in my `screen` and `track` event names?

> No, your `screen` and `track` event names must contain only alphanumeric characters. ActiveCampaign prescribes that the `screen` and `track` event names must contain only alphabets and numbers, and does not allow any special characters to be used.

Reviewer: ```suggestion
No, your `screen` and `track` event names must contain only alphanumeric characters. ActiveCampaign prescribes that the `screen` and `track` event names must contain only alphabets and numbers, and does not allow the usage of any special characters.
```
